examples/how_to/trackpy_local_max_peak_finder.py

```diff
@@ -0,0 +1,77 @@
+provides alternatives.
+
+:py:class:`.TrackpyLocalMaxPeakFinder` finds Gaussian spots by an implementation of the
+Crocker-Grier centroid-finding algorithm. Local maxima are treated as centroids and then the
+locations are refined to obtain sub-pixel accuracy. Unlike :py:class:`.LocalMaxPeakFinder`,
+:py:class:`.TrackpyLocalMaxPeakFinder` measures spot attributes like size and eccentricity but it
+also requires the user to manually set a minimum intensity threshold.
+
+Before running :py:class:`.TrackpyLocalMaxPeakFinder`, the image must be preprocessed to
+smooth noise and remove background. This can be done by setting ``preprocess=True`` or by running
+:py:class:`.Bandpass` filter separately. If done in :py:class:`.TrackpyLocalMaxPeakFinder`,
+``noise_size`` defines the width of the Gaussian blurring kernel for removing high frequency
+noise and ``smoothing_size`` defines the width of boxcar smoothing kernel. There is no formula
+for picking ``noise_size``, but larger values will lead to more blurring and it should be less
+than ``smoothing_size``. And ``smoothing_size`` should generally be the diameter of a spot
+rounded up to the nearest odd integer. The background should be set to zero with a ``threshold``
+set in  :py:class:`.Bandpass` or by
+ref:`clipping background to zero  <howto_clip_percentile_to_zero>`.
+
+For finding spots, ``spot_diameter`` is used to identify all possible spots and should be set to
+the diameter of the spots on zero background rounded up to the nearest odd integer. Then
+thresholds such as ``min_mass`` (integrated brightness), ``max_size`` (radius of gyration),
+``percentile`` (relative peak brightness), and ``separation`` (distance between spots) are used
+to filter out spots. Plotting these attributes can help ballpark cutoff values but :ref:`visually
+assessing <howto_spotfindingresults>` results is needed to validate the parameter settings before
+running in batch.
+
+.. warning::
+    :py:class:`.TrackpyLocalMaxPeakFinder` does not support ``is_volume = False``
+
+"""
+
+from starfish import data
+from starfish import FieldOfView
+from starfish.image import Filter
+from starfish.spots import FindSpots
+
+experiment = data.allen_smFISH(use_test_data=True)
+img = experiment['fov_001'].get_image(FieldOfView.PRIMARY_IMAGES)
+
+# filter to remove noise, remove background, blur, and clip
+bandpass = Filter.Bandpass(lshort=.5, llong=7, threshold=0.0)
+glp = Filter.GaussianLowPass(
+    sigma=(1, 0, 0),
+    is_volume=True
+)
+clip1 = Filter.Clip(p_min=50, p_max=100)
+clip2 = Filter.Clip(p_min=99, p_max=100, is_volume=True)
+clip1.run(img, in_place=True)
+bandpass.run(img, in_place=True)
+glp.run(img, in_place=True)
+clip2.run(img, in_place=True)
+
+
+tlmpf = FindSpots.TrackpyLocalMaxPeakFinder(
+    spot_diameter=5,  # must be odd integer
+    min_mass=0.02,
+    max_size=2,  # this is max radius
+    separation=7,
+    noise_size=0.65,  # this is not used because preprocess is False
```

Comment:
if it's not used, can we not specify this?

Reply:
I kept it because `noise_size` is mentioned in the text and is a reminder of what `preprocess` does. But okay, removed.